public/lib/helpers/timeline_helper.js

```diff
@@ -29,6 +31,60 @@ define(function (require) {
       }
     };
 
+    TimelineHelper.prototype.pluckLabel = function (hit, params, notify) {
+      let label = '';
+
+      let field;
+      if (params.labelFieldSequence) { // in kibi, we have the path property of a field
+        field = kibiUtils.getValuesAtPath(hit._source, params.labelFieldSequence);
+      } else {
+        field = _.get(hit._source, params.labelField);
+      }
+      if (field) {
+        if (this.isMultivalued(field)) {
+          notify.warning('Label field [' + params.labelField + '] is multivalued - the first value will be used.');
+        }
+        label = this.pickFirstIfMultivalued(field, '');
+      }
+
+      if (params.useHighlight) {
+        const fragmentCounts = {}; //key is fragment tag, value is count
+        Object.keys(hit.highlight).forEach(function (key) {
```

Comment:
here you could reused kibana's code for handling highlights, which I referenced above.